# model-volleyball: Machine Learning Models

In [1]:
# module imports
import pandas as pd

In [2]:
# import data
df = pd.read_csv('../data/full_matches.csv')

In [3]:
fr = df[df["season"] == "FR"]
so = df[df["season"] == "SO"]
jr = df[df["season"] == "JR"]
sr = df[df["season"] == "SR"]

#### Per-set totals for each season

In [4]:
print('FR season:')
fr_k_p_s = round(fr['kills'].sum() / fr['sets_played'].sum(), 2)
fr_a_p_s = round(fr['aces'].sum() / fr['sets_played'].sum(), 2)
fr_d_p_s = round(fr['digs'].sum() / fr['sets_played'].sum(), 2)
fr_b_p_s = round(fr['total_blocks'].sum() / fr['sets_played'].sum(), 2)

print(f'K/S: {fr_k_p_s} | A/S: {fr_a_p_s} | D/S: {fr_d_p_s} | B/S: {fr_b_p_s}')

FR season:
K/S: 1.7 | A/S: 0.73 | D/S: 3.22 | B/S: 0.07


In [5]:
print('SO season:')
so_k_p_s = round(so['kills'].sum() / so['sets_played'].sum(), 2)
so_a_p_s = round(so['aces'].sum() / so['sets_played'].sum(), 2)
so_d_p_s = round(so['digs'].sum() / so['sets_played'].sum(), 2)
so_b_p_s = round(so['total_blocks'].sum() / so['sets_played'].sum(), 2)

print(f'K/S: {so_k_p_s} | A/S: {so_a_p_s} | D/S: {so_d_p_s} | B/S: {so_b_p_s}')

SO season:
K/S: 2.55 | A/S: 0.97 | D/S: 3.02 | B/S: 0.07


In [6]:
print('JR season:')
jr_k_p_s = round(167 / jr['sets_played'].sum(), 2)
jr_a_p_s = round(120 / jr['sets_played'].sum(), 2)
jr_d_p_s = round(250 / jr['sets_played'].sum(), 2)
jr_b_p_s = round(8 / jr['sets_played'].sum(), 2)

print(f'K/S: {jr_k_p_s} | A/S: {jr_a_p_s} | D/S: {jr_d_p_s} | B/S: {jr_b_p_s}')

JR season:
K/S: 1.59 | A/S: 1.14 | D/S: 2.38 | B/S: 0.08


In [7]:
print('SR season:')
sr_k_p_s = round(sr['kills'].sum() / sr['sets_played'].sum(), 2)
sr_a_p_s = round(sr['aces'].sum() / sr['sets_played'].sum(), 2)
sr_d_p_s = round(sr['digs'].sum() / sr['sets_played'].sum(), 2)
sr_b_p_s = round(sr['total_blocks'].sum() / sr['sets_played'].sum(), 2)

print(f'K/S: {sr_k_p_s} | A/S: {sr_a_p_s} | D/S: {sr_d_p_s} | B/S: {sr_b_p_s}')

SR season:
K/S: 2.76 | A/S: 1.26 | D/S: 4.17 | B/S: 0.05


FR season:  
K/S: 1.7 | A/S: 0.73 | D/S: 3.22 | B/S: 0.07  

SO season:  
K/S: 2.55 | A/S: 0.97 | D/S: 3.02 | B/S: 0.07  

JR season:  
K/S: 1.59 | A/S: 1.14 | D/S: 2.38 | B/S: 0.08  

SR season:  
K/S: 2.76 | A/S: 1.26 | D/S: 4.17 | B/S: 0.05  
  
---
  
JR stats for aces and blocks seem to be par on the approximate development trajectory as compared to FR, SO, and SR. However, for digs and kills, the JR stats seems massively regressed -- especially compared with FR year. Odd, as I was also team captain for SO, JR, and SR, and played literally almost every set.  

1.59 kills/set for JR is substainally below FR and SO, and it is also odd that the very next year, it jumped up by 1.17 more kills? 2.38 digs/set for JR, which is also lower than both FR and SO, and also another huge jump of 1.79 digs.

#### Percent-change improvements between seasons

In [8]:
print('FR -> SO:')
k_pct = round(((so_k_p_s - fr_k_p_s) / fr_k_p_s) * 100, 2)
a_pct = round(((so_a_p_s - fr_a_p_s) / fr_a_p_s) * 100, 2)
d_pct = round(((so_d_p_s - fr_d_p_s) / fr_d_p_s) * 100, 2)
b_pct = round(((so_b_p_s -  fr_b_p_s) / fr_b_p_s) * 100, 2)

print(f'Kpct: {k_pct} | Apct: {a_pct} | Dpct: {d_pct} | Bpct: {b_pct}')

FR -> SO:
Kpct: 50.0 | Apct: 32.88 | Dpct: -6.21 | Bpct: 0.0


In [9]:
print('SO -> JR:')
k_pct = round(((jr_k_p_s - so_k_p_s) / so_k_p_s) * 100, 2)
a_pct = round(((jr_a_p_s - so_a_p_s) / so_a_p_s) * 100, 2)
d_pct = round(((jr_d_p_s - so_d_p_s) / so_d_p_s) * 100, 2)
b_pct = round(((jr_b_p_s - so_b_p_s) / so_b_p_s) * 100, 2)

print(f'Kpct: {k_pct} | Apct: {a_pct} | Dpct: {d_pct} | Bpct: {b_pct}')

SO -> JR:
Kpct: -37.65 | Apct: 17.53 | Dpct: -21.19 | Bpct: 14.29


In [10]:
print('JR -> SR:')
k_pct = round(((sr_k_p_s - jr_k_p_s) / jr_k_p_s) * 100, 2)
a_pct = round(((sr_a_p_s - jr_a_p_s) / jr_a_p_s) * 100, 2)
d_pct = round(((sr_d_p_s - jr_d_p_s) / jr_d_p_s) * 100, 2)
b_pct = round(((sr_b_p_s - jr_b_p_s) / jr_b_p_s) * 100, 2)

print(f'Kpct: {k_pct} | Apct: {a_pct} | Dpct: {d_pct} | Bpct: {b_pct}')

JR -> SR:
Kpct: 73.58 | Apct: 10.53 | Dpct: 75.21 | Bpct: -37.5


In [11]:
print('FR -> SR:')
k_pct = round(((sr_k_p_s - fr_k_p_s) / fr_k_p_s) * 100, 2)
a_pct = round(((sr_a_p_s - fr_a_p_s) / fr_a_p_s) * 100, 2)
d_pct = round(((sr_d_p_s - fr_d_p_s) / fr_d_p_s) * 100, 2)
b_pct = round(((sr_b_p_s - fr_b_p_s) / fr_b_p_s) * 100, 2)

print(f'Kpct: {k_pct} | Apct: {a_pct} | Dpct: {d_pct} | Bpct: {b_pct}')

FR -> SR:
Kpct: 62.35 | Apct: 72.6 | Dpct: 29.5 | Bpct: -28.57


FR -> SO:  
Kpct: 50.0 | Apct: 32.88 | Dpct: -6.21 | Bpct: 0.0  

SO -> JR:  
Kpct: -37.65 | Apct: 17.53 | Dpct: -21.19 | Bpct: 14.29  

JR -> SR:  
Kpct: 73.58 | Apct: 10.53 | Dpct: 75.21 | Bpct: -37.5  

FR -> SR:  
Kpct: 62.35 | Apct: 72.6 | Dpct: 29.5 | Bpct: -28.57
  
---
  
A negative drop between SO and JR year for kills and digs, despite me attending volleyball camps and striving to improve throughout the summer before the season? There is also another HUGE jump up from JR to SR with a 74% improvement for kills and a 75% improvement for digs. I don't think the numbers show the true statistics for my JR season. 
  
Formula used: `((b-a) / a ) * 100`

#### MACHINE LEARNING